public/pages/CreateDetector/components/ConfigureFieldMapping/components/RequiredFieldMapping/FieldNameSelector.tsx

```diff
@@ -15,43 +14,57 @@ interface SIEMFieldNameProps {
 }
 
 interface SIEMFieldNameState {
-  selectedOption?: string;
+  selectedOptions: EuiComboBoxOptionOption<string>[];
   errorMessage?: string;
 }
 
 export default class FieldNameSelector extends Component<SIEMFieldNameProps, SIEMFieldNameState> {
   constructor(props: SIEMFieldNameProps) {
     super(props);
+
+    let selectedOptions = [] as EuiComboBoxOptionOption<string>[];
+    if (props.selectedField) {
+      selectedOptions = [
+        {
+          label: props.selectedField,
+        },
+      ];
+    }
+
     this.state = {
-      selectedOption: props.selectedField,
+      selectedOptions: selectedOptions,
     };
   }
 
-  onChange: React.ChangeEventHandler<HTMLSelectElement> = (
-    event: ChangeEvent<HTMLSelectElement>
-  ) => {
-    this.setState({ selectedOption: event.target.value });
-    this.props.onChange(event.target.value);
+  onMappingChange = (selectedOptions: EuiComboBoxOptionOption<string>[]) => {
+    if (!selectedOptions.length) selectedOptions = [...this.state.selectedOptions];
+    this.setState({ selectedOptions });
+    this.props.onChange(selectedOptions[0]?.label);
   };
 
   render() {
-    const { isInvalid } = this.props;
+    const { selectedOptions } = this.state;
+    const { isInvalid, fieldNameOptions } = this.props;
+
+    const comboOptions = fieldNameOptions.map((option) => ({
+      value: option,
+      label: option,
+    }));
+
     return (
       <EuiFormRow
         style={{ width: '100%' }}
         isInvalid={isInvalid}
         error={isInvalid ? 'Name already used' : undefined}
```

Comment:
To clarify, would this be the only error message, or is it possible for the combo box to be empty? I see that you've added `isClearable={false}`, but I'm just wondering about cases where the field mapper may not be able to assume the appropriate mapping, and the page loads with an empty combo box. This might not be an issue anymore, but thought I'd bring it up as an edge case. Would it make sense to add some logic like the following:
```
let errorMessage = '';
if (!selectedOptions.length) {
    errorMessage = 'Select a field.';
} else {
    errorMessage = isInvalid ? 'Field already used.' : undefined;
}
```
As a side note, we always want to end error messages with `.`; we treat it as a statement essentially so it should have punctuation.

Reply:
@amsiglan is it allowed for mappings to be empty? If yes, then let me change the validation and update the PR...

Reply:
@AWSHurneyt Fixed the punctuation and did refactoring allowing the mappings to have empty fields, waiting for @amsiglan to confirm and this is good to go.

Reply:
Sorry for the confusion. I don't believe we want to have empty field mappings; I'm more thinking of cases where the page might load without autofilling the combo boxes. We would want to display an error message calling out that it's necessary to select an option.

Reply:
Right, so the message is there only if the user makes the same mapping twice, not if there is no mapping.

Reply:
We are aiming to refactor this component such that the mapping will be empty on load but I don't think we should show an error message for that, just marking it as required is enough and also the status icon should be set to warning until the mapping is complete

Reply:
Let's circle around this in a follow up task where we also move this component to Define Detector step